packages/react-components/src/components/RichTextEditor/Plugins/CopyPastePlugin.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { PluginEvent, EditorPlugin } from 'roosterjs-editor-types';
+import { CompatiblePluginEventType, CompatiblePasteType } from 'roosterjs-editor-types-compatible';
+
+/**
+ * CopyPastePlugin is a plugin for handling copy and paste events in the editor.
+ */
+export default class CopyPastePlugin implements EditorPlugin {
+  getName(): string {
+    return 'CopyPastePlugin';
+  }
+
+  initialize(): void {}
+
+  dispose(): void {}
+
+  onPluginEvent(event: PluginEvent): void {
+    removeImageElement(event);
+  }
+}
+
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export const removeImageElement = (event: PluginEvent): void => {
+  // We don't support the pasting options such as paste as image yet.
+  if (event.eventType === CompatiblePluginEventType.BeforePaste && event.pasteType === CompatiblePasteType.Normal) {
+    event.fragment.querySelectorAll('img').forEach((image) => {
+      // If the image is the only child of its parent, remove all the parents of this img element before removing the image element.
+      let parentNode = image.parentElement;
+      const parentsToRemove: Array<HTMLElement | null> = [];
+      while (parentNode?.childNodes.length === 1) {
+        parentsToRemove.push(parentNode);
+        parentNode = parentNode.parentElement;
+      }
+      parentsToRemove.forEach((parent) => {
```

Comment:
I think we can jut remove the top most parent node without going though all items (we probably don't even need to remove the image node in this case...)

Reply:
Deleting the parent node doesn't delete it's children with it 
![image](https://github.com/Azure/communication-ui-library/assets/107075081/3473fc79-eb87-457b-8f03-c188256eaba9)
And the fragment doesn't have remove all children api so deleting from the top most layer wouldn't work.

Reply:
Can we use https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren#emptying_a_node?

In case, if it's not possible, can we remove the nodes in the same loop? Something like:
```
      let parentNode = image.parentElement;
      let newParent = parentNode;
      while (parentNode?.childNodes.length <= 1) {
        newParent = parentNode.parentElement;
        parentNode.remove();
        parentNode = newParent;
      }
```

Reply:
`And the fragment doesn't have remove all children api so deleting from the top most layer wouldn't work.`
Sorry, For  the top most parent  I meant the last one that is returned in your while loop

Reply:
Hey @Leah-Xia-Microsoft,

I just tested
```
export const removeImageElement = (event: PluginEvent): void => {
  // We don't support the pasting options such as paste as image yet.
  if (event.eventType === CompatiblePluginEventType.BeforePaste && event.pasteType === CompatiblePasteType.Normal) {
    event.fragment.querySelectorAll('img').forEach((image: HTMLElement) => {
      // If the image is the only child of its parent, remove all the parents of this img element before removing the image element.
      let parentNode = image.parentElement;
      let previousParent = image;
      while (parentNode?.childNodes.length === 1) {
        previousParent = parentNode;
        parentNode = parentNode.parentElement;
      }
      previousParent?.remove();
    });
  }
};

```
all tests pass and nodes are removed as expected

Reply:
Please note that I use `previousParent` not  `parentNode` as `parentNode` may be undefined if the last node has no `parentElement`. Just wondering if that's a situation that happened for the shared screenshot (the parent is undefined and there was nothing to remove)